backend/src/services/integration-auth/integration-sync-secret.ts

```diff
@@ -3631,7 +3631,16 @@ const syncSecretsBitBucket = async ({
   const res: { [key: string]: BitbucketVariable } = {};
 
   let hasNextPage = true;
-  let variablesUrl = `${IntegrationUrls.BITBUCKET_API_URL}/2.0/repositories/${integration.targetEnvironmentId}/${integration.appId}/pipelines_config/variables`;
+
+  let variablesUrl: string;
+
+  if (integration.targetServiceId) {
+    // scope: deployment environment
+    variablesUrl = `${IntegrationUrls.BITBUCKET_API_URL}/2.0/repositories/${integration.targetEnvironmentId}/${integration.appId}/deployments_config/environments/${integration.targetServiceId}/variables`;
+  } else {
+    // scope: repository
+    variablesUrl = `${IntegrationUrls.BITBUCKET_API_URL}/2.0/repositories/${integration.targetEnvironmentId}/${integration.appId}/pipelines_config/variables`;
+  }
```

Comment:
upon updating a value in the source... it never seems to completely sync to the destination because an error gets thrown
<img width="900" alt="image" src="https://github.com/user-attachments/assets/7645d13b-9942-433b-8f6a-356889813394">

Can you look into this as well? A few users have reported the same behavior

Reply:
good catch! will look into this